### PR TITLE
add merge operator for SignalProducer

### PR DESCRIPTION
**Pull request by larryonoff**
It often usable to use merge operator for SignalProducer. `Signal` already has merge operator.

### CODE REVIEW

File: ReactiveCocoa/Swift/Flatten.swift [hunk 1]
```diff
@@ -552,6 +552,22 @@ extension SignalType {
...
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public static func merge<S : SequenceType where S.Generator.Element == SignalProducer<Value, Error>>(producers: S) -> SignalProducer<Value, Error> {
+		return SignalProducer(values: producers).flatten(.Merge)
+	}
+	
+	/// Merges the given producers into a single `SignalProducer` that will emit all values
+	/// from each of them, and complete when all of them have completed.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public static func merge<Value, Error: ErrorType>(producers: SignalProducer<Value, Error>...) -> SignalProducer<Value, Error> {
+		return SignalProducer(values: producers).flatten(.Merge)
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Flatten.swift]: This could just call the above variant and pass `producers`.

File: ReactiveCocoa/Swift/Flatten.swift [hunk 2]
```diff
@@ -552,6 +552,22 @@ extension SignalType {
...
+	/// from each of them, and complete when all of them have completed.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public static func merge<S : SequenceType where S.Generator.Element == SignalProducer<Value, Error>>(producers: S) -> SignalProducer<Value, Error> {
+		return SignalProducer(values: producers).flatten(.Merge)
+	}
+	
+	/// Merges the given producers into a single `SignalProducer` that will emit all values
+	/// from each of them, and complete when all of them have completed.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public static func merge<Value, Error: ErrorType>(producers: SignalProducer<Value, Error>...) -> SignalProducer<Value, Error> {
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Flatten.swift]: Can you add a `...` variant like this to `Signal` as well if we're going to try to even out the interfaces?

**Comment by mdiep** [comment]
Thanks for the PR! I agree this is a handy thing to have.

**Comment by larryonoff** [comment]
Thank you for the comments! I've updated this PR according to your comments.

**Comment by larryonoff** [comment]
What do you think about the case when `combineLatest` and `zip` operators are free functions, when `merge` is not.

Maybe it's better to move `merge` operator from protocols?

**Comment by ikesyo** [comment]
> What do you think about the case when `combineLatest` and `zip` operators are free functions, when `merge` is not.
> 
> Maybe it's better to move `merge` operator from protocols?

@larryonoff That should be discussed in #2926.

**Comment by larryonoff** [comment]
@ikesyo thanks!

File: ReactiveCocoaTests/Swift/SignalProducerSpec.swift [hunk 3]
```diff
@@ -1893,6 +1893,71 @@ class SignalProducerSpec: QuickSpec {
 				expect(started) == false
 			}
 		}
+		
+		describe("SignalProducer.merge") {
```
Comment (ikesyo) [inline ReactiveCocoaTests/Swift/SignalProducerSpec.swift]: I think this (and `describe("Signal.merge")` in `SignalSpec.swift`) should be defined in `FlattenSpec.swift`.

Comment (larryonoff) [inline ReactiveCocoaTests/Swift/SignalProducerSpec.swift]: Will move it.

Comment (larryonoff) [inline ReactiveCocoaTests/Swift/SignalProducerSpec.swift]: Done

**Comment by larryonoff** [comment]
`Signal.merge()` and `SignalProducer.merge()` moved into FlattenSpec.

**Comment by mdiep** [comment]
This is great! Thanks for the PR. ✨

**Pull request merged**